v0/destinations/hs/HSTransform-v2.js

```diff
@@ -277,63 +291,87 @@ const batchIdentify = (
 
     let batchEventResponse = defaultBatchRequestConfig();
 
+    if (batchOperation === "create") {
+      batchEventResponse.batchedRequest.endpoint = BATCH_IDENTIFY_CRM_CREATE_NEW_CONTACT;
+    } else if (batchOperation === "update") {
+      batchEventResponse.batchedRequest.endpoint = BATCH_IDENTIFY_CRM_UPDATE_NEW_CONTACT;
+    }
+
     if (batchOperation === "create") {
       // create operation
       chunk.forEach(ev => {
-        // format properties into batch structure
-        // eslint-disable-next-line no-param-reassign
-        ev.message.body.JSON.properties = getCRMUpdatedProps(
-          ev.message.body.JSON.properties
-        );
-
-        // duplicate email can cause issue with create in batch
-        // updating the existing one to avoid duplicate
-        // as same event can fire in batch one of the reason
-        // can be due to network lag or processor being busy
-        const isDuplicate = identifyResponseList.find(data => {
-          return (
-            data.properties.email === ev.message.body.JSON.properties.email
-          );
-        });
-        if (isDefinedAndNotNullAndNotEmpty(isDuplicate)) {
-          // array is being shallow copied hence changes are affecting the original reference
-          // basically rewriting the same value to avoid duplicate entry
-          isDuplicate.properties = ev.message.body.JSON.properties;
+        // if source is of rETL
+        if (ev.message.source === "rETL") {
+          identifyResponseList.push({ ...ev.message.body.JSON });
+          batchEventResponse.batchedRequest.endpoint = `${ev.message.endpoint}/batch/create`;
```

Comment:
nit: probably move this to line 295  and use ternary operator? first glance looked like we are using contact for all `create` operations and later realized this is overriding!

Reply:
done